AKAlarm/AppDelegate.swift

```diff
@@ -8,57 +8,106 @@
 
 import UIKit
 import CoreData
+import UserNotifications
 
 @UIApplicationMain
-class AppDelegate: UIResponder, UIApplicationDelegate {
-
+class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate{
+    
     var window: UIWindow?
-
-
+    
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
+        
+        UNUserNotificationCenter.current().delegate = self
+        UNUserNotificationCenter.current().requestAuthorization(options:[.badge, .alert, .sound]) {
+            (granted, error) in
+            
+            guard error == nil else {
+                print(error?.localizedDescription ?? "Error in registering")
+                return
+            }
+            if granted {
+                application.registerForRemoteNotifications()
```

Comment:
it seems this method could be something else as we are not using remote notificitons